### DISCUSSION TITLE
Is an RTU_RTU_Bridge_Filtered possible?

### DISCUSSION THREAD

**Discussion by CRachfahl** [eModbus/eModbus#379] [Q&A] [answered]
Hi everyone,

I'm new to ESP32 programming and what to implement a RTU to RTU Bridge with a filter on one ESP32 board and have some questions:

1. Is this possible with eModbus?
2. What ESP32 should I use?
3. any examples already out there?

Anyway thanks for your the great project

Carsten

**Comment by Miq1** [comment, marked as answer]
It is, in fact. There is a set of functions exactly for that purpose in the bridge code:
```
// Add/remove request/response filters
  bool addRequestFilter(uint8_t aliasID, MBSworker rF);
  bool removeRequestFilter(uint8_t aliasID);
  bool addResponseFilter(uint8_t aliasID, MBSworker rF);
  bool removeResponseFilter(uint8_t aliasID);
```
The ``aliasID`` can be a dedicated server id to be filtered or ``ANY_SERVER``.

For a RTU/RTU bridge you will need an ESP32 providing 2 Serial interfaces, so any "classic" WROOM or WROVER will do. 

There is an [example for a WiFi/RTU filtered bridge](https://github.com/eModbus/eModbus/tree/master/examples/WiFi-RTU_Bridge_Filtered) that basically is the same with another bridge server-side protocol, but that does not matter.

**Comment by CRachfahl** [comment]
Thanks @Miq1 I will try to code it and if I have question I hope I can come back.

Carsten

**Reply by Miq1** [reply]
Of course! Natürlich! 😉

**Reply by CRachfahl** [reply]
Another German? Grüsse aus dem schönen sonnigen Sauerland ;-)
